### PR TITLE
Closes #12708 -- Clarify split_to_map documentation

### LINKED ISSUE #12708: split_to_map clarification in documentation
Just discovered `split_to_map`, excellent for mapping my Spark work back to Presto when needed!

However, as I noted for Spark/Hive:

https://github.com/apache/hive/pull/553

https://github.com/apache/spark/pull/23888

The documentation can lead/has led to some confusion. Hive/Spark's `str_to_map` uses _regular expression_ delimiters, but, IINM, [Presto's implementation uses fixed character matching](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/scalar/SplitToMapFunction.java#L64):

```
int entryEnd = string.indexOf(entryDelimiter, entryStart);
// ...
int keyEnd = keyValuePair.indexOf(keyValueDelimiter);
```

This is fine, of course -- and in fact, I probably prefer the Presto version -- but I think that given the difference vs. Spark/Hive, it definitely needs to be documented more precisely, esp. since `split_to_map` was explicitly suggested as an homage to `str_to_map` in #2544

### PR DESCRIPTION
**Pull request by MichaelChirico**
#12708 Straightforward PR.

Should something be added to the release notes?

Let me know any other touch-up needed.

### CODE REVIEW

File: presto-docs/src/main/sphinx/functions/string.rst [hunk 1]
```diff
@@ -134,6 +136,8 @@ String Functions
     containing an array of values for each unique key. ``entryDelimiter`` splits ``string``
     into key-value pairs. ``keyValueDelimiter`` splits each pair into key and value. The
     values for each key will be in the same order as they appeared in ``string``.
+    Note that ``entryDelimiter`` and ``keyValueDelimiter`` are
```
Comment (highker) [inline presto-docs/src/main/sphinx/functions/string.rst]: Keep the same length of this line as the previous one.

**Comment by MichaelChirico** [comment]
@mayankgarg1990 thanks for the comment, it makes sense. How about:

> Note that ``entryDelimiter`` and ``keyValueDelimiter`` are interpreted literally, i.e., as full string matches.

People with a keen eye will recognize `literal` is often used as a contrast to `regex` so the meaning would still be conveyed, & `full string matches` provides further clarification for readers who may not know exactly what `literal` interpretation is.

**Comment by mayankgarg1990** [comment]
@MichaelChirico - that sounds reasonable to me

**Comment by MichaelChirico** [comment]
@mayankgarg1990 thanks for the feedback; pushed the change

**Comment by MichaelChirico** [comment]
@mayankgarg1990 done

**Comment by mayankgarg1990** [comment]
Thanks for the extra documentation!

**Comment by MichaelChirico** [comment]
HTH in any small way given I use Presto every day!

**Comment by highker** [comment]
Next time please make sure the commit message is clean. `Closes #12708 -- Clarify split_to_map documentation` is by no means a good message title.

**Comment by MichaelChirico** [comment]
@higherk that is the opposite of helpful feedback. would rather you said nothing.

**Pull request merged**